core/src/test/java/io/confluent/rest/SslTest.java

```diff
@@ -226,6 +227,38 @@ public void testHttpsOnly() throws Exception {
     }
   }
 
+  @Test
+  public void test_WhenCalledWithRestrictedCipher_ConnFails() throws Exception {
+    TestMetricsReporter.reset();
+    Properties props = new Properties();
+    String uri = "https://localhost:8080";
+    props.put(RestConfig.LISTENERS_CONFIG, uri);
+    props.put(RestConfig.METRICS_REPORTER_CLASSES_CONFIG, "io.confluent.rest.TestMetricsReporter");
+    configServerKeystore(props);
+    // Restrict HTTPs Server to cipher TLS_AES_128_GCM_SHA256
+    props.put(RestConfig.SSL_CIPHER_SUITES_CONFIG, "TLS_AES_128_GCM_SHA256");
+    TestRestConfig config = new TestRestConfig(props);
+    SslTestApplication app = new SslTestApplication(config);
+    try {
+      app.start();
+
+      // Correct Cipher, returns 200.
+      // TLSv1.3 should be specified with TLS_AES_128_GCM_SHA256 only for test-setup, as this cipher is specific to TLSv1.3.
+      // NOTE - the behaviour of restricting cipher in REST is agnostic of TLS protocols.
+      int statusCode = makeGetRequest(uri + "/test",
+          clientKeystore.getAbsolutePath(), SSL_PASSWORD, SSL_PASSWORD, "TLSv1.3", "TLS_AES_128_GCM_SHA256");
+      assertEquals(200, statusCode, EXPECTED_200_MSG);
+      assertMetricsCollected();
+      // Incorrect Cipher, throws an exception.
```

Comment:
could you also add tests where client specifies the right cipher but tls version < 1.3?

Reply:
So there are 3 cases w.r.t ciphers
1. Request fails as cipher is invalid, due to incompatible tls protocol.
2. Request fails as Cipher is invalid, as its not part of the restricted ciphers on the server.
3. Request succeeds, as Cipher is valid, compatible with tls protocol, and part of restricted cipher.

2. & 3. are captured here.

> could you also add tests where client specifies the right cipher but tls version < 1.3?

I think this case 1. But isn't this case unrelated/independent to config `ssl.cipher.suite`, so outside the scope of the original ticket, wdyt?

Reply:
Anyways i have added test-case for 1 as its easy enough.